### PR TITLE
build-style/cmake.sh: add generic do_check function.

### PR DESCRIPTION
**Pull request by maxice8**
https://cmake.org/Wiki/CMake/Testing_With_CTest

### CODE REVIEW

File: common/build-style/cmake.sh [hunk 1]
```diff
@@ -47,6 +47,24 @@ do_build() {
 	${make_cmd} ${makejobs} ${make_build_args} ${make_build_target}
 }
 
+do_check() {
+	if [ -z "$make_cmd" ] && [ -z "$make_check_target" ]; then
+		if make -q test 2>/dev/null; then
+			:
+		else
```
Comment (Vaelatern) [inline common/build-style/cmake.sh]: `if ! make -q test 2>/dev/null; then`

Comment (maxice8) [reply common/build-style/cmake.sh]: why ?

Comment (Vaelatern) [reply common/build-style/cmake.sh]: Save the useless 2 lines of confusion

Comment (chneukirchen) [reply common/build-style/cmake.sh]: NACK. This will mingle $? = 1 and ? = 2

Comment (chneukirchen) [reply common/build-style/cmake.sh]: Please revert.

**Comment by chneukirchen** [comment]
As explained, this is broken. `! make` will never set `$? -eq 2`. Please fix.

**Pull request closed without merging**